### PR TITLE
[WIP] Review Docs in light of config changes

### PR DESCRIPTION
**Pull request by wxtim**
- References Changes summarized in https://github.com/cylc/cylc-flow/issues/3696.
- Also includes adding basic references to Platform settings.
- Replaces references to suite with refs to workflows.

I've added suggestion comments in places where I see the possibility of making changes which I believe may spark debate.

Pre-requisites:
* https://github.com/cylc/cylc-flow/pull/3968
* https://github.com/cylc/cylc-flow/pull/3963

### CODE REVIEW

**Comment by wxtim** [comment]
@dpmatthews @oliver-sanders @hjoliver  - this PR is still in draft form, but I'm tagging you so that you can see I'm having a good look at this stuff.

It's by no means finished atm.

File: src/glossary.rst [hunk 1]
```diff
@@ -4,28 +4,35 @@ Glossary
 .. glossary::
    :sorted:
 
-   suite
-   Cylc suite
-      A Cylc suite is a directory containing a :cylc:conf:`flow.cylc` file
+   workflow
+   Cylc workflow
+      A Cylc workflow is a directory containing a :cylc:conf:`flow.cylc` file
```
Comment (wxtim) [inline src/glossary.rst:9]: This isn't really what a workflow is, but how it's defined.
```suggestion
A Cylc workflow is a way of organizing one or more programs, specifying relationships between them. Is is defined in a directory containing a :cylc:conf:`flow.cylc` file.
```

File: src/glossary.rst [hunk 2]
```diff
@@ -4,28 +4,35 @@ Glossary
...
       which contains :term:`graphing<graph>` representing a workflow.
 
-   suite directory
-      The suite directory contains all of the configuration for a suite e.g.
+   workflow directory
+      The workflow directory contains all of the configuration for a workflow e.g.
       the :cylc:conf:`flow.cylc` file.
```
Comment (wxtim) [inline src/glossary.rst:15]:
```suggestion
     The workflow directory contains all of the configuration for a workflow. 
     In its most basic form the workflow directory will contain a :cylc:conf:`flow.cylc` file.
```

**Comment by hjoliver** [comment]
Did you intend to close this @wxtim ?

**Comment by wxtim** [comment]
> Did you intend to close this @wxtim ?

Yep. It suffered massive mission creep

**Pull request closed without merging**